### PR TITLE
Add prosthetics

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Add some prosthetics."

#### Purpose of change
With the new addition of lost limbs, prosthetics were the next logical step of additions to counteract this. To do this, I decided to add them.

#### Describe the solution
Add two new prosthetic items that can act as replacement limbs for characters without them, those being the left and right legs. These rely on mutations and enchantments to properly add the limbs, which have been defined in JSON. This PR also includes two dummy mutations that keep players from wearing a prosthetic limb if they are not missing one. These can be scavenged from medical-related locations and some corpse loot.

#### Describe alternatives you've considered
Not doing this, or doing it all in C++

#### Testing
Went into the game and started as the para-amputee. I could properly wear prosthetic limbs and work with them, letting me walk around and travel unhindered. When I took it off, I had the proper limb stub again. When I didn't start as the para-amputee and tried to wear them, I was properly hindered.

#### Additional context
Currently, the prosthetic limbs are slightly less efficient than ones made of flesh, from what I could gather from my studies to be accurate. If this is incorrect, I'll change it. Also, I don't know what all goes into prosthetic production, so I don't have a recipe ready for it. I could at it in this PR if that's preferred, or at a later date.

I have plans to add variants of prosthetics to the game, and different "grades" of them, ranging from the antiquated to the far future, possibly including bionic limbs that the Exodii might sell.

~This will need #68289 to merge so that I can get the prosthetics to act the way I want them to.~

### CODE REVIEW

File: data/json/items/armor/legs_prosthetics.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+[
+  {
+    "id": "right_leg_prosthetic",
+    "type": "ARMOR",
+    "name": { "str_sp": "prosthetic right leg" },
+    "description": "A prosthetic leg, used as a replacement limb for those missing one.  It's modern design and sleek ergonomics make it very comfortable and highly suitable for use, and 3-D designing make look exactly like the original.  This one is for the right leg.",
```
Comment (TheShadowFerret) [inline data/json/items/armor/legs_prosthetics.json:6]:
```suggestion
    "description": "A prosthetic leg, used as a replacement limb for those missing one.  Its modern design and sleek ergonomics make it very comfortable and highly suitable for use, and 3D designing makes it look exactly like the original.  This one is for the right leg.",
```

File: data/json/items/armor/legs_prosthetics.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "material_thickness": 1,
+    "flags": [ "LEG_PROSTHETIC" ],
+    "relic_data": { "passive_effects": [ { "has": "WORN", "condition": "ALWAYS", "mutations": [ "RIGHT_PROSTHETIC_LEG" ] } ] },
+    "armor": [ { "encumbrance": 0, "coverage": 100, "covers": [ "leg_stub_r", "leg_r" ] } ]
+  },
+  {
+    "id": "left_leg_prosthetic",
+    "type": "ARMOR",
+    "name": { "str_sp": "prosthetic left leg" },
+    "description": "A prosthetic leg, used as a replacement limb for those missing one.  It's modern design and sleek ergonomics make it very comfortable and highly suitable for use, and 3-D designing make look exactly like the original.  This one is for the left leg.",
```
Comment (TheShadowFerret) [inline data/json/items/armor/legs_prosthetics.json:26]:
```suggestion
    "description": "A prosthetic leg, used as a replacement limb for those missing one.  Its modern design and sleek ergonomics make it very comfortable and highly suitable for use, and 3D designing makes it look exactly like the original.  This one is for the left leg.",
```

File: data/json/mutations/mutations.json [hunk 3]
```diff
@@ -310,17 +310,79 @@
...
+    "restricts_gear": [ "leg_stub_r" ]
+  },
+  {
+    "type": "mutation",
+    "id": "NO_RIGHT_LEG",
+    "name": { "str": "No Right Leg" },
+    "points": 0,
+    "valid": false,
+    "purifiable": false,
+    "description": "Wether by a serious accident, birth defect, or other means, you have lost your right leg.  You can hobble around, but will likely need a wheelchair or other means of locomotion to get somewhere far.",
```
Comment (TheShadowFerret) [inline data/json/mutations/mutations.json:342]:
```suggestion
    "description": "Whether by a serious accident, birth defect, or other means, you have lost your right leg.  You can hobble around, but will likely need a wheelchair or other means of locomotion to get somewhere far.",
```

File: data/json/mutations/mutations.json [hunk 4]
```diff
@@ -310,17 +310,79 @@
...
+    "enchantments": [ "ENCH_NO_RIGHT_LEG" ]
+  },
+  {
+    "type": "mutation",
+    "id": "NO_LEFT_LEG",
+    "name": { "str": "No Left Leg" },
+    "points": 0,
+    "valid": false,
+    "purifiable": false,
+    "description": "Wether by a serious accident, birth defect, or other means, you have lost your left leg.  You can hobble around, but will likely need a wheelchair or other means of locomotion to get somewhere far.",
```
Comment (TheShadowFerret) [inline data/json/mutations/mutations.json:356]:
```suggestion
    "description": "Whether by a serious accident, birth defect, or other means, you have lost your left leg.  You can hobble around, but will likely need a wheelchair or other means of locomotion to get somewhere far.",
```

**Review by TheShadowFerret — COMMENTED** [review]
If you wear clothing over prosthetics and take off your legs, does the clothing fall off?

**Comment by PatrikLundell** [comment]
An advanced 3D printed leg might look very close to a real one on the person it was made for (skin color, leg shape, etc. not to mention length and proportions), but wouldn't for a random needy scavenger. A problem with 3D printed parts is that they don't have a lot of adjustments in them as they're made to fit their target exactly, so a less advanced design might actually be better for a scavenger as it is more likely to be adjustable.

**Comment by MNG-cataclysm** [comment]
> An advanced 3D printed leg might look very close to a real one on the person it was made for (skin color, leg shape, etc. not to mention length and proportions), but wouldn't for a random needy scavenger. A problem with 3D printed parts is that they don't have a lot of adjustments in them as they're made to fit their target exactly, so a less advanced design might actually be better for a scavenger as it is more likely to be adjustable.

I was thinking that the part was just 3D modeled to get the proportions and shape right. Should I clarify this in the description, or make a new variant?

**Comment by Zireael07** [comment]
Let's go with a simple solution for now, we can always add variants later (I really want to play my wheelchair challenge start!)

**Comment by PatrikLundell** [comment]
I'd probably skip the 3D part and just say that it's shaped like a real leg (as opposed to the "purely functional" ones that do not have the cosmetics, but only the "bone" parts).

**Comment by MNG-cataclysm** [comment]
> If you wear clothing over prosthetics and take off your legs, does the clothing fall off?

It used to, but I fixed that now.

**Comment by PatrikLundell** [comment]
It's really more complicated with clothing... Pants shouldn't be affected as there's just going to be some flopping at the bottom (although one could make an argument for an increase in encumbrance, as the flopping can catch on things, etc.), but shoes and stuff attached to the leg below the connection point (ankle sheath, for instance) wouldn't have anything to attach to anymore. The same would go for arms, once they are added. No hovering wrist watches...

Oh, just thought of another complication: taking damage to a prosthetic limb shouldn't cost HP to the limb, but rather damage the prosthetic.

**Comment by MNG-cataclysm** [comment]
> Oh, just thought of another complication: taking damage to a prosthetic limb shouldn't cost HP to the limb, but rather damage the prosthetic.

I set it so that prosthetics cover 100% of the limb their provide, so any strikes to them will always damage the prosthetic.

**Comment by lispcoc** [comment]
I personally made a prosthetics mod before. I think you need to be careful that your HP does not recover when you put it on or take it off.

**Comment by MNG-cataclysm** [comment]
> I personally made a prosthetics mod before. I think you need to be careful that your HP does not recover when you put it on or take it off.

I'll have to figure that out, and that might need some C++ to make it work. Would it be alright if I took a look at your mod for some reference material?

**Comment by lispcoc** [comment]
> I'll have to figure that out, and that might need some C++ to make it work. Would it be alright if I took a look at your mod for some reference material?

https://github.com/lispcoc/RoboticLimb
Although Japanese is included, I don't think you need any knowledge of Japanese if you just want to understand EOC. HP information is saved by generating a separate item for each player's HP, but this is not a very wise method, so I think it would be better to implement some kind of function in C++.

**Comment by MNG-cataclysm** [comment]
After some poking around, I've realized that I'm going to need some help with the C++ regarding this PR for the prosthetics to work like I want them to.

**Comment by MNG-cataclysm** [comment]
@lispcoc Would you be willing to provide an example of item EOCs and how they affect things, please? I'd like to cross-reference something when I start working on the new code for limb modeling.

**Comment by lispcoc** [comment]
> @lispcoc Would you be willing to provide an example of item EOCs and how they affect things, please? I'd like to cross-reference something when I start working on the new code for limb modeling.

There are still few EOCs that can be applied to items, but I think this test data will be useful as a usage example.
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/mods/TEST_DATA/EOC.json

**Comment by Maleclypse** [comment]
Tested and there are some weird edge cases like being able to bandage a damaged prosthetic leg and it heals the faux limb but since it doesn't repair damage to the item, I'm going to assume that what was damaged was the remaining portion of the flesh leg where the prostheses attaches.  Everything else I tested worked as expected.
